test/persistence/mzcompose.py

```diff
@@ -166,3 +167,28 @@ def workflow_compaction(c: Composition) -> None:
     c.rm("materialized", "testdrive-svc", destroy_volumes=True)
 
     c.rm_volumes("mzdata")
+
+
+# Run tests using the S3 implementation of Blob.
+#
+# TODO: the kafka-sources test has a failpoints test that specifically use a file
+# blob that is a no-op in this context. The test happens to succeed because it
+# checks for recovery from failpoint, but doesn't actually assert that the failpoint
+# itself did anything, and we're able to write to Kafka faster than we can ingest
+# data into persist at the moment. The proper fix is to move backend specific
+# failpoints somewhere else.
+def workflow_s3(c: Composition) -> None:
+    seed = round(time.time())
+    tests = {
+        "kafka-sources": workflow_kafka_sources,
+        "user-tables": workflow_user_tables,
+    }
+
+    for test_name, test_fn in tests.items():
+        materialized = Materialized(
+            options=f"{mz_options} --persist-storage-enabled --persist-storage=s3://mtlz-test-persist-1d-lifecycle-delete/persist-s3-test-{test_name}-{seed}",
```

Comment:
nit: lets make this prefix named something more obviously connected to this composition. maybe `test-persistence-mzcompose`? another idea is `mzcompose-persistence-s3` to mirror the cli invocation of `./mzcompose --find persistence run s3`

Reply:
hmm I like `mzcompose-persistence-s3`